plugins/webapi/faucet/plugin.go

```diff
@@ -82,10 +68,7 @@ func requestFunds(c echo.Context) error {
 		}
 	}
 
-	faucetPayload, err := faucet.NewRequest(addr, config.Node().Int(faucet.CfgFaucetPoWDifficulty), accessManaPledgeID, consensusManaPledgeID)
-	if err != nil {
-		return c.JSON(http.StatusBadRequest, jsonmodels.FaucetResponse{Error: err.Error()})
-	}
+	faucetPayload := faucet.NewRequest(addr, accessManaPledgeID, consensusManaPledgeID, request.Nonce)
```

Comment:
can we check that `faucetPayload` is valid with respect to `request.Nonce`? Otherwise there is no point in submitting it to the tangle, no?